doc/articles/routed-events.md

```diff
@@ -0,0 +1,179 @@
+| Any local handlers?   no--------+    +-------^--------------+
++-------yes-------------+         |            |
+         |                        |    [10]----+--------------+
+[4]------v--------------+         |    | Event is bubbling    |
+| Invoke local handlers |         |    | to parent in         <--+
++--------+--------------+         |    | managed code (Uno)   |  |
+         |                        |    +-------^--------------+  |
+[5]------v--------------+         |            |                 |
+| Is the event handled  |         v    [6]-----no-------------+  |
+| by local handlers?    no------------>| Event is coming from |  |
++-------yes-------------+              | platform?            |  |
+         |                             +------yes-------------+  |
+[9]------v--------------+                      |                 |
+| Any parent interested |              [7]-----v--------------+  |
+| by this event?        yes-+          | Is the event         |  |
++-------no--------------+   |          | bubbling natively?   no-+
+         |                  |          +------yes-------------+
+[12]-----v--------------+   |                  |
+| Processing finished   |   v          [8]-----v--------------+
+| for this event.       |  [10]        | Event is returned    |
++-----------------------+              | for native           |
+                                       | bubbling in platform |
+                                       +----------------------+
+```
+
+ 1. **An event is fired**: when an event is intercepted from the platform.
+ 2. **Event dispatcher**: the source element in visual tree receive the event through its event handler.
+ 3. **Local handlers?**: check if there is any local handlders for the event.
+ 4. **Invoke handlers**: they are invoked one after the other, taking care of the "IsHandled" status.
+ 5. **Handled?**: check if any of the local handlers marked the event as _handled_.
+ 6. **Originating from platform?**: check if the source of the event is from native code.
+ 7. **Bubbling natively?**: check if the bubbling should be done natively for this event.
+    bubbling natively will let controls not implemented in Uno the ability to intercept the event.
+ 8. **Returned for native bubbling**: let the event bubbling in the platform.
+ 9. **Any Parent Interested?**: check if any parent control is interested by a "handled" version of this event
+    (using `AddEventHandler(<evt>, <handler>, handledEventsToo: true)`).
+ 10. **Bubbling in managed code**: propagate the event in Uno to parents (for events not bubbling natively
+    or `handledEventsToo: true`).
+ 11. **Parent defined?**: if the element is connected to any parent element.
+ 12. **Processing finished**: no more handlers is interested by this event. Propagation is stopped.
+    Native bubbling is stopped too, because the event is fully handled.
+
+## Native bubbling vs Managed bubbling
```

Comment:
It's unfortunate that consumers have to worry about this distinction IMO. It's probably acceptable for 'RoutedEvents v1', but ideally we would find a way to have native views in the tree (which is pretty common) and still support handledEventsToo.

Reply:
I agree...  but we're limited by the fact we are dealing with platforms where this feature doesn't exist.
Maybe we'll find another way to support it in the future.